File: app/src/main/java/com/doapps/habits/listeners/EmptyListListener.java
package com.doapps.habits.listeners;

import java.util.Observable;


public class EmptyListListener extends Observable {
    public static final EmptyListListener listener = new EmptyListListener();
    private boolean mChanged;

    public void isEmpty(boolean status) {
        mChanged = status;
        if (mChanged) {
            super.notifyObservers();
        }
    }

    @Override
    public synchronized boolean hasChanged() {
        return mChanged;
    }

}

File: app/src/main/java/com/doapps/habits/listeners/EmptyListListener.kt
package com.doapps.habits.listeners

import java.util.*


class EmptyListListener : Observable() {
  private var mChanged = false

  fun isEmpty(status: Boolean) {
    mChanged = status
    if (mChanged) {
      super.notifyObservers()
    }
  }

  override fun hasChanged(): Boolean {
    synchronized(this) {
      return mChanged
    }
  }

  companion object {
    val listener = EmptyListListener()
  }
}